src/server/scripts/EasternKingdoms/zone_isle_of_queldanas.cpp

```diff
@@ -677,6 +679,119 @@ class npc_greengill_slave : public CreatureScript
     };
 };
 
+/*######
+## npc_irespeaker
+######*/
+
+
+enum IreSpeaker_Spells
+{
+    SPELL_FEL_FIREBALL = 35913,
+    SPELL_CURSE_OF_WEAKNESS = 18267,
+    SPELL_FEL_CONSUMPTION = 45023,
+};
+
+
+enum IreSpeaker_Events
+{
+    EVENT_FEL_FIREBALL = 1,
+    EVENT_CURSE_OF_WEAKNESS = 2,
+};
+
+
+class npc_irespeaker : public CreatureScript
+{
+public:
+    npc_irespeaker() : CreatureScript("npc_irespeaker") { }
+
+    struct npc_irespeakerAI : public ScriptedAI
+    {
+        npc_irespeakerAI(Creature* creature) : ScriptedAI(creature), isInMeleeRange(false) { }
+
+        EventMap events;
+        bool isInMeleeRange;
+
+        void Reset() override
+        {
+            events.Reset();
+            isInMeleeRange = false;
+            me->SetReactState(REACT_PASSIVE);  // Start passive to prevent charging
+        }
+
+        void JustEngagedWith(Unit* who) override
+        {
+            if (who)
+            {
+                me->SetReactState(REACT_PASSIVE);  // Stay passive until in melee range
+                events.ScheduleEvent(EVENT_FEL_FIREBALL, 5000);  // Schedule Fel Fireball every 5-30 seconds
+                isInMeleeRange = false;  // Start in ranged mode
+                me->Attack(who, true);  // Ensure the NPC attacks the target
```

Comment:
This is not needed, attackstart is handled on its own function

Reply:
Thank you for your feedback. I am just getting back into working in the TC/AzerothCore environment.

Reply:
@Nyeriah Regarding the new "creature registering way". Are you refering to something like this?
RegisterKarazhanCreatureAI(boss_bigbadwolf); ?


That doesnt exists for the Zone Isle of Quel'Dalas right now.